Core/Lib/Models/ResultListModel.cs

```diff
@@ -12,12 +12,14 @@ public ResultListModel()
         public ResultListModel(IEnumerable<TValue> value)
         {
             Value = value;
+            Size = value.Count();
         }
         public ResultListModel(TValue[] value)
         {
             Value = value;
+            Size = value.Length;
         }
 
-        public int Size { get => Value.Count(); }
+        public int Size { get; set; }
```

Comment:
setter should be private, because you set the value in the constructor.

Reply:
This is a default, but can be overridden from outside (like when paging the size could be all items, but the Value may only have 10 items). Maybe we should be more clear on this, or have a property valuesize and totalsize to make it more clear what these properties are for? @Johanh-od , @cosmoro , any suggestions here?

Reply:
I should treat Size as a property that can't be set from outside the instance. The Size refers to the current retrieved record count. In case of paging this is the size of a page (or possible less if it's the last page). For the other paging info (total record count, total pages, current page, page size) I should create a new class containing this info and add this as a property of the ResultListModel

Reply:
In my opinion, Size and Value.Count() are different things. Size is the the total size of a result, since the Value contains only the paginated items.
See below how it's used now in DataService.

public static async Task<ResultListModel<TModel>> ToResultList<TEntity, TModel>(this IQueryable<TEntity> entities, QueryModel queryModel, IMapper mapper)
            where TEntity : class, IIdEntity
        {
            var result = entities.ProjectTo<TModel>(mapper.ConfigurationProvider);

            return new ResultListModel<TModel>
            {
                Value = queryModel.NoLimit
                    ? await result.ToListAsync()
                    : await result.Skip(queryModel.Offset).Take(queryModel.Limit).ToListAsync(),
                Size = await result.CountAsync()
            };
        }

Reply:
@cosmoro , to be more clear on the intent of a property, maybe how you use it we call it TotalSize, and have Size reflect the  actual size of the value. Would that work for you?

Reply:
@cosmoro agree that Size and Value.Count() are different things. We could improve the naming on this part though, but I suggested to separate the paging from the result object, because not all GET calls do have paging (e.g. if you only retrieve one record).

Reply:
So, we could also create an PagedResultListModel<TModel>. This could derive from ResultListModel<TModel> but have additional properties like TotalSize, PageSize, PageIndex, PageNumber, SortingProperty and SortingDirection. Just that the Filter will not know about that, you would explicitly have to return a PagedResultListModel. But then again, that could be a nice static method on ResultModel, just like there are 2 of those already. An example of a controller could look like this what is in the attachement
[HoursController.txt](https://github.com/lensbysits/core-backend/files/9131109/HoursController.txt)
.

Reply:
Of course, since it's only a default and can be overridden from outside, that's fine. 
Adding other extra properties would be fine, but renaming the property as totalsize, for instance, will affect our whole project since we use a lot this property.

Reply:
Hi @cosmoro , @Johanh-od, I added a ResultPagedListModel and an extra property on the list-size. I think this is backwards compatible, but improves future clarity on what the intent is (also commented some properties). Can this make it into develop? A sample of the usage is in this controller:
[HoursController.txt](https://github.com/lensbysits/core-backend/files/9131543/HoursController.txt), where I'm faking the QueryModel, that would usually come from the front-end of course (action parameter [FromQuery]QueryModel queryModel)).